tests/E2E/test/selectors/BO/customers/customer.js

```diff
@@ -10,21 +10,21 @@ module.exports = {
     password_input: '//*[@id="customer_password"]',
     days_select: '//*[@id="customer_birthday_day"]',
     month_select: '//*[@id="customer_birthday_month"]',
-    years_select: '//*[@id="customere_birthday_year"]',
+    years_select: '//*[@id="customer_birthday_year"]',
     save_button: '//*[@id="main-div"]//button[contains(text(),"Save")]',
-    customer_filter_by_email_input: '//*[@id="form-customer"]//input[@name="customerFilter_email"]',
-    email_address_value: '//*[@id="form-customer"]//td[%ID]',
+    customer_filter_by_email_input: '//*[@id="customer_email"]',
+    email_address_value: '//*[@id="customer_filter_form"]//tbody//td[%ID]',
     reset_button: '//*[@id="table-customer"]//button[@name="submitResetcustomer"]',
-    edit_button: '//*[@id="form-customer"]//a[@title="Edit"]',
-    dropdown_toggle: '//*[@id="form-customer"]//button[@data-toggle="dropdown"]',
-    delete_button: '//*[@id="form-customer"]//a[@title="Delete"]',
-    delete_first_option: '//*[@id="deleteMode_real"]',
-    delete_second_option: '//*[@id="deleteMode_deleted"]',
-    delete_confirmation_button: '//*[@id="content"]//input[@value="Delete"]',
-    select_customer: '//*[@id="form-customer"]//input[@name="customerBox[]"]',
-    bulk_actions_button: '//*[@id="bulk_action_menu_customer"]',
-    bulk_actions_delete_button: '//*[@id="form-customer"]//div[contains(@class,"bulk-actions")]//a[contains(@onclick,"submitBulkdeletecustomer")]',
-    empty_list_icon: '//*[@id="table-customer"]//div[contains(@class,"list-empty-msg")]',
+    edit_button: '//*[@id="customer_grid_table"]//a[@data-original-title="Edit"]',
+    dropdown_toggle: '//*[@id="customer_grid_table"]//a[@data-toggle="dropdown"]',
+    delete_button: '//*[@id="customer_grid_table"]//a[contains(@class,"js-delete-customer-row-action")]',
+    delete_first_option: '//input[@id="delete_customers_delete_method_0"]',
+    delete_second_option: '//input[@id="delete_customers_delete_method_1"]',
+    delete_confirmation_button: '//*[@id="customer_grid_delete_customers_modal"]//button[contains(@class, "js-submit-delete-customers")]',
+    select_customer: '//table[@id="customer_grid_table"]//td[contains(@class, "bulk_action-type")]/div/label',
+    bulk_actions_button: '//button[contains(@class, "js-bulk-actions-btn")]',
+    bulk_actions_delete_button: '//button[@id="customer_grid_bulk_action_delete_selection"]',
+    empty_list_icon: '//*[@id="customer_grid_table"]//tbody//td/div/p[2]',
```

Comment:
How can we make this selector cleaner?

Reply:
This element is not easily targetable: no ID, no special class or anything that would make the selector cleaner. I could be able to transform it into a CSS one: it might **look** cleaner but that won't change the fact that it's kinda hard to target.

Reply:
I understand that, my question was more "can we (the core team) do something to make this selector cleaner?"
If you could show  screenshot of the element you want to target, we may add an attribute on it.

Reply:
![image](https://user-images.githubusercontent.com/49909275/56752360-5de4ce00-6788-11e9-85f6-d27a7af2c2cc.png)
Oh my bad I didn't understand. That would be really cool ! 
This selector targets the checkbox in front of every customer, that you use when you want to perform bulk actions.

Reply:
Hmm, for this one it looks like you can rely on the class `js-bulk-action-checkbox`, doesn't it?

In this part of code, I was talking about the line 27 

```js
empty_list_icon: '//*[@id="customer_grid_table"]//tbody//td/div/p[2]',
```
which seems really tricky.

Reply:
I can't target the checkbox itself because it's not visible and Selenium won't let me interact with it, I have to target the whole visible element which is the "label" tag.
For the second one, there was an ID on the image before. I could remove some of the selectors in the second part (like "td/div") but since I'm looking for the "No records found" icon I know there's only one td and one div in the table.
This is what the markups looks like (and I'm only interested in the "p" tag):
![no_records_found](https://user-images.githubusercontent.com/49909275/56790748-abebe700-6805-11e9-8333-361fc5f6d65a.png)
I thought about using the "mb-2" class but IMHO it's too generic and I can't be sure there won't be another one in the page or even in the table.

Reply:
Can you have a look at https://github.com/PrestaShop/PrestaShop/pull/13570/commits/0ba6582432806c372e5d2b9a3b6b1a4d8db57392 included in this PR ? I added a class on the parent div to help you target the empty state.

Reply:
That's perfect.

Reply:
@Quetzacoalt91 I modified the selector to use your new class. Test is OK.
Thanks again ! :ok_hand: